src/server/registry/catalog/catalog.go

```diff
@@ -14,19 +14,117 @@
 
 package catalog
 
-import "net/http"
+import (
+	"encoding/json"
+	"fmt"
+	ierror "github.com/goharbor/harbor/src/internal/error"
+	"github.com/goharbor/harbor/src/pkg/repository"
+	reg_error "github.com/goharbor/harbor/src/server/registry/error"
+	"github.com/goharbor/harbor/src/server/registry/util"
+	"net/http"
+	"sort"
+	"strconv"
+)
 
 // NewHandler returns the handler to handler catalog request
-func NewHandler() http.Handler {
-	return &handler{}
+func NewHandler(repoMgr repository.Manager) http.Handler {
+	return &handler{
+		repoMgr: repoMgr,
+	}
 }
 
-type handler struct{}
+type handler struct {
+	repoMgr repository.Manager
+}
 
+// ServeHTTP ...
 func (h *handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	// TODO implement
+	switch req.Method {
+	case http.MethodGet:
+		h.get(w, req)
+	}
+}
+
+func (h *handler) get(w http.ResponseWriter, req *http.Request) {
+	var maxEntries int
+	var err error
+
+	reqQ := req.URL.Query()
+	lastEntry := reqQ.Get("last")
+	withN := reqQ.Get("n") != ""
+	if withN {
+		maxEntries, err = strconv.Atoi(reqQ.Get("n"))
+		if err != nil || maxEntries < 0 {
+			err := ierror.New(err).WithCode(ierror.BadRequestCode).WithMessage("the N must be a positive int type")
+			reg_error.Handle(w, req, err)
+			return
+		}
+	}
+
+	var repoNames []string
+	// get all repositories
+	// ToDo filter out the untagged repos
+	total, repoRecords, err := h.repoMgr.List(req.Context(), nil)
+	if err != nil {
+		reg_error.Handle(w, req, err)
+		return
+	}
+	if total <= 0 {
+		h.sendResponse(w, req, repoNames)
+		return
+	}
+	for _, r := range repoRecords {
+		repoNames = append(repoNames, r.Name)
+	}
+	sort.Strings(repoNames)
+	if !withN {
+		h.sendResponse(w, req, repoNames)
+		return
+	}
```

Comment:
This will cause the API returns all repositories when `last` in the query but `n` not in it.

Reply:
we will not handle the query only with `last`